### ISSUE TITLE
Proxy startup time too long, causes commands to time out

### ISSUE THREAD

**Issue by jinnovation** [jinnovation/kele.el#151]
Commands like `kele-get` that rely on proxy processes for given contexts can take too long to start up the proxy. This can result in the command timing out and silently failing. At this point, the user has to do `kele-get` again, at which point the command will succeed (presumably related to the initial failure).

Symptoms include the initial failure resulting in a kubeconfig update:
```
Pulling kubeconfig contents...done
Waiting for kubeconfig update to finish...done
```

Note that while we refer to `kele-get` specifically here this issue impacts all commands that use proxy processes, e.g. `kele-list`.

We have various tickets open to address different aspects of this issue, e.g. #72.

## Definition of Done

- Proxy-based commands like `kele-get` execute successfully on first try

**Comment by jinnovation** [comment]
I wonder if this is caused by `kubectl`'s invalidation of the discovery cache that happens on first `kubectl` invocation after 10-minute expiry.

**Comment by jinnovation** [comment]
One possible solution here is to have `kele--proxy-process` start the proxy asynchronously via, say, `async-start-process`. With this, we can effectively turn `kele--context-proxy-ledger` into a ledger of proxy **promises** and defer the readiness/liveliness checks on the proxy API to right when the user actually needs the proxy.

**Comment by jinnovation** [comment]
Given that we fix #72 with #153, closing this as well under assumption that underlying issue is now fixed.
